Fixing keyboard focus to see all Lessons element of issue #17205

Rahul-lalwani-learner opened the pull request:
…ghting with a box on focusing while we want underline

## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #17205 
2. This PR does the following: I have changed the "see all lessons" span tag to the anchor tag in Line numbers 152-157 in order to resolve the anomaly that occurred. Initially, the element of the span tag was highlighted within the box which was not expected behavior. Then on line number 152, I have also added the class "nav-item-bottom-link" to set the color same as the default element. I have also added href="/learn/math" so that when the user press enters it will redirect the user to the correct page.  Finally in the same line, I have added tabindex=0 to bring keyboard focus to "See all lessons" Element

## Essential Checklist

- [X] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [X] The linter/Karma presubmit checks have passed locally on your machine.
- [X] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [X] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->
#### Proof that changes are correct
Before: 

https://user-images.githubusercontent.com/82160150/218687286-a4b8ad8f-2853-43d5-ad5e-8205cf8e0d9e.mp4

After: 

https://user-images.githubusercontent.com/82160150/219529055-1a0c8d97-ee5a-466a-a9dd-40dcca1960d6.mp4

#### Proof of changes on desktop with slow/throttled network

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

Rahul-lalwani-learner committed:
changed see all lesson span tag with anchor tag because it was highlighting with a box on focusing while we want underline

Rahul-lalwani-learner committed:
Merge branch 'develop' into fixing-keyboard-focus

Rahul-lalwani-learner committed:
Merge branch 'develop' into fixing-keyboard-focus

Rahul-lalwani-learner committed:
Merge branch 'develop' into fixing-keyboard-focus

Rahul-lalwani-learner committed:
Merge branch 'develop' into fixing-keyboard-focus

Rahul-lalwani-learner commented:
hey!! @seanlip 
can you help me with this pr when the first time I updated the branch all jobs were marked as successful but now every time different jobs are failing. This time frontend job failed and I am not able to understand the mistake in my code.

seanlip commented:
@Rahul-lalwani-learner This might be a dev workflow issue -- looping in @gp201 @U8NWXD to advise.

gp201 commented:
@Rahul-lalwani-learner The error log given below is a known error:
https://github.com/oppia/oppia/actions/runs/4184277567/jobs/7249895104
We're currently working on a fix for that. https://github.com/oppia/oppia/pull/17245

I have restarted the frontend tests.

The lighthouse error https://github.com/oppia/oppia/actions/runs/4177140010/jobs/7234341319 is also a known error. The error is being tracked here: https://github.com/oppia/oppia/issues/16762 . We're working on a fix for that too.

Rahul-lalwani-learner commented:
@SanjaySajuJacob All the Jobs are Passed. PTAL

U8NWXD commented:
Deassigning myself since the jobs have all passed now

Rahul-lalwani-learner committed:
Merge branch 'develop' into fixing-keyboard-focus

Rahul-lalwani-learner committed:
Merge branch 'develop' into fixing-keyboard-focus

Rahul-lalwani-learner commented:
@SanjaySajuJacob plz review this merge as soon as possible cause with every pull merge new jobs are tends to fail.

Rahul-lalwani-learner commented:
@seanlip can you help me with this pr review, any way to contact @SanjaySajuJacob

SanjaySajuJacob commented on core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.html:
```diff
@@ -149,12 +149,12 @@
                     <div (keydown)="onMenuKeypress($event, 'learnMenu', {enter: ACTION_CLOSE})"
                          class="nav-item-bottom-link">
-                      <span class="underline">
+                      <a class="underline nav-item-bottom-link" tabindex="0">
```
Hey @Rahul-lalwani-learner, the underline seems to be visible but would you confirm that when focus reaches the  `see all Lessons` element, on clicking enter, user is redirected to the appropriate page as well. If this does not occur, would please add this functionality as well and update the proof for that in the PR as well. Thanks!

Rahul-lalwani-learner replied:
I have changed it as you asked me please let me know if there is something more on that.

SanjaySajuJacob commented:
Hey @Rahul-lalwani-learner I have added my review, please take a look and feel free to ping me if you need some clarity.
Also, once you have assigned a reviewer to the PR, give them 24 hours to respond before you ping them again. Thanks!

Rahul-lalwani-learner committed:
Adding href to anchor tag in See all Lessons

by adding href in anchor tag of See all lessons in line number 152 now can press enter to redirect to desired page when focus is on see all lessons element

Rahul-lalwani-learner commented on core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.html:
```diff
@@ -149,12 +149,12 @@
                     <div (keydown)="onMenuKeypress($event, 'learnMenu', {enter: ACTION_CLOSE})"
                          class="nav-item-bottom-link">
-                      <span class="underline">
+                      <a class="underline nav-item-bottom-link" tabindex="0" href="/learn/math">
```
@SanjaySajuJacob I have added href="/learn/math" so that when the user press enters on see all lessons it will redirect it to the correct page. i think this is what you were asking from me to do

Rahul-lalwani-learner commented:
@SanjaySajuJacob PTAL, do the changes I did correctly as you required them. I have changed in file where I have added href="/learn/math" to redirect to all lessons page as you asked and I have also updated my PR as per changes. I also changed the after video in the "Proof all changes are correct" section so you can also take a look to see if everything is correct if it required more changes

Rahul-lalwani-learner committed:
Merge branch 'develop' into fixing-keyboard-focus
